src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -344,6 +360,35 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
             if (controllerData.xrController) {
                 controllerGlobalPosition = controllerData.xrController.pointer.position;
                 controllerData.xrController.getWorldPointerRayToRef(controllerData.tmpRay);
+                const hand = controllerData.xrController.inputSource.hand;
+                if(hand)
+                {
+                    const xrJoint = hand.get("index-finger-tip");
+                    if(xrJoint)
+                    {
+                        let pose = _xrFrame.getJointPose!(xrJoint, this._xrSessionManager.referenceSpace);
+                        if (!pose || !pose.transform) {
+                            return;
+                        }
+                        const pos = pose.transform.position;
+                        const originalMesh = SphereBuilder.CreateSphere("ParentSphere", { diameter: 1 });
+                        originalMesh.scaling.set(0.01, 0.01, 0.01);
+
+                        // populate information for near pick
+                        const pickRadius = (pose.radius || 0.008)/2;
+                        controllerData.pickMesh = originalMesh.createInstance("IndexPickSphere");

```

Comment:
I know the webXRHandTracker holds information about the joint radius somewhere... Can't we just use a reference to the index tip joint mesh itself here, and the radius information that comes with it?

Reply:
love this idea, not sure how to do that though. let me dig some and see

Reply:
we are using that radius information infact. "pose.radius"